homeassistant/components/light/hyperion.py

```diff
@@ -90,34 +140,107 @@ def turn_on(self, **kwargs):
         """Turn the lights on."""
         if ATTR_RGB_COLOR in kwargs:
             self._rgb_color = kwargs[ATTR_RGB_COLOR]
-        else:
+            self._rgb_mem = kwargs[ATTR_RGB_COLOR]
+        elif self._rgb_mem == [0, 0, 0]:
             self._rgb_color = self._default_color
+            self._rgb_mem = self._default_color
+        else:
+            self._rgb_color = self._rgb_mem
```

Comment:
You should not change any state holding instance attributes in the service methods. Those will be updated in `update` after the service call, since this is a polling entity.

Reply:
I agree, I will update in a commit in a second